scripts/debugging_scripts/check_eth_status.py

```diff
@@ -0,0 +1,196 @@
+
+"""
+Usage:
+    scripts/debugging_scripts/check_eth_status.py
+
+Description:
+    Check status on the ethernet cores
+"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from ttexalens.context import Context
+from ttexalens.device import Device, OnChipCoordinate
+from ttexalens.register_store import read_word_from_device
+
+from check_per_device import run as get_check_per_device
+
+from ttexalens.hw.tensix.blackhole.blackhole import BlackholeDevice
+from ttexalens.hw.tensix.wormhole.wormhole import WormholeDevice
+from scripts.debugging_scripts import utils
+from triage import ScriptConfig, log_check, run_script
+
+script_config = ScriptConfig(
+    depends=["check_per_device"],
+)
+
+
+@dataclass
+class EthCoreDefinitions:
+    port_status: int
+    retrain_count: int
+    rx_link_up: int
+    heartbeat: int
+    mailbox: int
+    mailbox_slots: int
+
+
+class EthCore(ABC):
+    """
+    Base class for Ethernet cores that provides common functionality.
+    """
+
+    def __init__(self, location: OnChipCoordinate, context: Context):
+        self.location = location
+        self.context = context
+        self.eth_core_definitions = EthCoreDefinitions(**self.get_addresses())
+
+    @abstractmethod
+    def get_addresses(self) -> dict:
+        """Return architecture-specific addresses as a dictionary."""
+        pass
+
+    @abstractmethod
+    def port_status_to_string(self, port_status: int) -> str:
+        """Convert port status value to a readable string."""
+        pass
+
+    def check_for_heartbeat(self) -> tuple[bool, str]:
+        """Check for heartbeat at the heartbeat address."""
+        previous_data = 0
+        # Check for a changing value at the heartbeat address. Read up to 100 times
+        for i in range(100):
+            read_data = read_word_from_device(self.location, self.eth_core_definitions.heartbeat, context=self.context)
+            if read_data != previous_data:
+                return True, ""
+            previous_data = read_data
+        return False, f"No heartbeat detected for {self.location.to_user_str()}"
+
+    def get_results(self):
+        """Get and log all ethernet core status results."""
+        # HEARTBEAT
+        passed, message = self.check_for_heartbeat()
+        log_check(passed, message)
+
+        # PORT STATUS
+        if self.eth_core_definitions.port_status is not None:
+            port_status = read_word_from_device(
+                self.location, self.eth_core_definitions.port_status, context=self.context
+            )
+            port_status_str = self.port_status_to_string(port_status)
+            if port_status_str == None:
+                log_check(
+                    False, f"{self.location.to_user_str()} port_status is unknown: {port_status} ({port_status_str})"
+                )
+            else:
+                utils.INFO(f"{self.location.to_user_str()} port_status: {port_status} ({port_status_str})")
```

Comment:
Once you go to tt-triage data visualization:
```suggestion
            log_check(
                port_status_str is not None,
                f"{self.location.to_user_str()} port_status is unknown: {port_status} ({port_status_str})"
            )
            # Port status should be returned and visualized by tt-triage instead of console prints
```